src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckCodeFixProvider.cs

```diff
@@ -53,23 +53,27 @@ private static ExpressionSyntax GetCondition(SyntaxNode node)
                     return ((WhileStatementSyntax)node).Condition;
                 case SyntaxKind.IfStatement:
                     return ((IfStatementSyntax)node).Condition;
+                case SyntaxKind.ReturnStatement:
+                    return ((ReturnStatementSyntax)node).Expression;
+                case SyntaxKind.LocalDeclarationStatement:
+                    return ((LocalDeclarationStatementSyntax)node).Declaration.Variables[0].Initializer.Value;
```

Comment:
Currently we pick the "leftmost" initializer/expression to simplify definitive assignment check. This can be extended later when we analyze the "flow" rather than just matching a "pattern".